servers/republik/graphql/resolvers/_mutations/updateNewsletterSubscription.js

```diff
@@ -1,11 +1,34 @@
-const { ensureSignedIn, Roles } = require('@orbiting/backend-modules-auth')
+const { Roles } = require('@orbiting/backend-modules-auth')
 
 module.exports = async (_, args, context) => {
-  const { req, user, user: me, t, mail } = context
-  const { updateNewsletterSubscription, errors } = mail
-  ensureSignedIn(req)
-  const { name, subscribed, status } = args
-  Roles.ensureUserIsMeOrInRoles(user, me, ['supporter, admin'])
+  const {
+    userId,
+    name,
+    subscribed,
+    status
+  } = args
+
+  const {
+    user: me,
+    pgdb,
+    req,
+    t,
+    mail: {
+      updateNewsletterSubscription,
+      errors
+    }
+  } = context
+
+  const user = userId
+    ? await pgdb.public.users.findOne({ id: userId })
+    : me
```

Comment:
So, if I enter a invalid (read non-existent) user id I just implicitly update my own prefs? Is this intentional?